### DISCUSSION TITLE
Material Canvas

### DISCUSSION THREAD

**Discussion by jeremyong-az** [o3de/sig-graphics-audio#56] [Ideas]
Thread for discussion about ongoing efforts related to material canvas, tooling, and code generation

**Comment by santorac** [comment]
Here's a link to an RFC from @gadams3 for a Material Canvas tool https://github.com/o3de/sig-graphics-audio/issues/51

**Comment by mekhontsev** [comment]
Here's a link to an RFC  for a Material Canvas shader generation [#57](https://github.com/o3de/sig-graphics-audio/issues/57)

**Reply by gadams3** [reply]
Thank you! I also added appendices to the material canvas RFC with notes that were made while writing the RFC about templates, output notes, code generation, and inserting the data into the node and slot configurations. Some of the things described are already in place. I'm currently working on setting up test templates for code generation, based on those notes, using minimal PBR as a starting point. In addition to that, I've implemented general support for auto saving and hot reloading across the tools. Having discussions to eliminate issues and make sure that works consistently for materials, material types, shaders, and models. We will need to look into AP iteration times compiling shaders, which will affect seeing previews between iterations.
